src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -138,15 +141,28 @@ public static bool IsPathRooted(string path)
         // only contains whitespace characters an ArgumentException gets thrown.
         public static string GetPathRoot(string path)
         {
-            if (path == null) return null;
+            if (path == null)
+                return null;
+
             if (PathInternal.IsEffectivelyEmpty(path))
                 throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
 
-            // Need to return the normalized directory separator
             path = PathInternal.NormalizeDirectorySeparators(path);
```

Comment:
Would it work to call GetPathRoot first, and only normalize the directory separators in what GetPathRoot returns?

Reply:
Probably... The only case I think might be broken is UNCs, and in a pretty corner case. `\\Server\\Share`, `\\Server\\\Share`, etc. would no longer work. Not sure that is worth worrying about.

Reply:
i can try that tomorrow

Reply:
I tried reversing the two operations and none of the existing tests failed for getpathroot()...should we keep the original one or
``` ReadOnlySpan<char> result = GetPathRoot(path.AsReadOnlySpan());
            if (PathInternal.AreSame(path, result))
                return PathInternal.NormalizeDirectorySeparators(path);
           
            return PathInternal.NormalizeDirectorySeparators(new string(result));
```

Reply:
Yes, that's ok. You shouldn't need the AreSame here as the span one doesn't change the contents and only slices from the start. Just check the length.